fairlearn/reductions/_exponentiated_gradient/_lagrangian.py

```diff
@@ -149,7 +149,12 @@ def _call_oracle(self, lambda_vec):
                                          constant=redY_unique[0])
             self.n_oracle_calls_dummy_returned += 1
         else:
-            classifier = pickle.loads(self.pickled_estimator)
+            # use sklearn.base.clone to clone the estimator.
+            # It is the same as copy.deepcopy for non-sklearn estimators (By using safe=False).
+            # For sklearn estimators, it is more efficient as it would not
+            # do a copy.deepcopy. Rather, instantiate a new estimator using the
+            # get_params() internally.
+            classifier = clone(estimator=self.estimator, safe=False)
```

Comment:
Please rename this to estimator while you're at it. Classifier indicates that this can't cover other tasks, which I've tried to fix with renaming before, but evidently I missed this one.

Reply:
Sure 👍